### PR TITLE
2022 10 stream download disable too big

### PR DESCRIPTION
**Pull request by aurel-l**
## Purpose
- Disable download when file is too big https://www.ebi.ac.uk/panda/jira/browse/TRM-28312
- Fix download URLs https://www.ebi.ac.uk/panda/jira/browse/TRM-28347

## Approach
- Disable download:
  - Hardcode limit (will need to update if it changes)
  - Disable download link (points to nowhere, click doesn't close panel, explanation in title attribute)
  - Disable URL generation for stream (replace URL with comment-style message), disable copy too
- Fix download URLs
  - wrap user query in parenthesis before combining with query from facets
    - Note that this was a problem when the user query was in the style of 'a OR b' plus a facet 'c' because it would generate 'a OR b AND c' whereas now it does '(a OR b) AND c'.

## Testing
- Manual testing with different big and small queries
- Manual checking if the parenthesis were correctly added in the generated URLs and in the download link

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/shared/components/download/Download.tsx [hunk 1]
```diff
@@ -279,12 +281,20 @@ const Download: FC<DownloadProps> = ({
...
         </Button>
+        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
         <a
-          href={downloadUrl}
-          className="button primary"
+          href={downloadCount > DOWNLOAD_SIZE_LIMIT ? undefined : downloadUrl}
+          className={cn('button', 'primary', {
+            disabled: downloadCount > DOWNLOAD_SIZE_LIMIT,
+          })}
+          title={
```
[the remark this answers is not in the document]
Comment (aurel-l) [reply src/shared/components/download/Download.tsx:290]: Not really as this is part of the default browser behaviour with title tags. We could use the tooltip library to do this, but if we wanted to go this way we'd need to do that at the Button level in Franklin in order to have consistent behaviour throughout the website.

File: src/shared/components/download/DownloadAPIURL.tsx [hunk 2]
```diff
@@ -72,25 +75,40 @@ const DownloadAPIURL = ({
...
-      </section>
+      <div className={disableStream ? helper.disabled : undefined}>
+        <h4>API URL {isStreamEndpoint && ' using the streaming endpoint'}</h4>
+        {isStreamEndpoint &&
+          'This endpoint is resource-heavy but will return all requested results.'}
+        <CodeBlock lightMode>
+          {disableStream ? (
+            <>
+              {
+                '// the streaming endpoint is unavailable for queries of more than '
```
[the remark this answers is not in the document]
Comment (aurel-l) [reply src/shared/components/download/DownloadAPIURL.tsx:92]: I'm worried users would still go and copy the URL themselves from here even if we disable the copy button. Using this comment text here was also an easy way for me to explain what is going on without relying entirely on the title attributes that users might miss

[the remark this answers is not in the document]
Comment (aurel-l) [reply src/shared/components/download/DownloadAPIURL.tsx:92]: I could also still render the URL but blocking selection with CSS in order to make it complicated to the user to copy the URL (not impossible though)

File: src/shared/components/download/DownloadAPIURL.tsx [hunk 3]
```diff
@@ -72,25 +75,40 @@ const DownloadAPIURL = ({
...
+          ) : (
+            apiURL
+          )}
+        </CodeBlock>
+        <section className="button-group">
+          <Button
+            variant="primary"
+            className={styles['copy-button']}
+            onClick={() => handleCopyURL(apiURL)}
+            disabled={disableStream}
```
[the remark this answers is not in the document]
Comment (aurel-l) [reply src/shared/components/download/DownloadAPIURL.tsx]: Added a title now, and adressed some of the suggested changes. Do have a look again now

File: src/shared/components/download/DownloadAPIURL.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlrice) [reply src/shared/components/download/DownloadAPIURL.tsx]: Sorry, I wasn't very clear in my comment. I meant to keep it as you had it - (not display the URL and keeping the `'// the streaming endpoint is unavailable ...'`) but not applying the disabled styling to the code block. My thinking was that someone may be less inclined to read something that is disabled.

[the remark this answers is not in the document]
Comment (aurel-l) [reply src/shared/components/download/DownloadAPIURL.tsx]: Ah, right, I understand now, I here what I have at this point

**Review by dlrice — APPROVED** [review]

**Pull request merged**